plugins/generator-1.18.2/forge-1.18.2/templates/dispensebehaviour.java.ftl

```diff
@@ -35,34 +35,39 @@
 
 package ${package}.item.extension;
 
+<#compress>
 @Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD) public class ${name}ItemExtension {
     	@SubscribeEvent public static void init(FMLCommonSetupEvent event) {
     		event.enqueueWork(() -> DispenserBlock.registerBehavior(${mappedMCItemToItem(data.item)}, new OptionalDispenseItemBehavior() {
     			public ItemStack execute(BlockSource blockSource, ItemStack stack) {
+    				<#assign hasSuccessCondition = hasProcedure(data.dispenseSuccessCondition)>
     				ItemStack itemstack = stack.copy();
     				Level world = blockSource.getLevel();
     				Direction direction = blockSource.getBlockState().getValue(DispenserBlock.FACING);
     				int x = blockSource.getPos().getX();
     				int y = blockSource.getPos().getY();
     				int z = blockSource.getPos().getZ();
 
+    				<#if hasSuccessCondition>
     				this.setSuccess(<@procedureOBJToConditionCode data.dispenseSuccessCondition/>);
+    				</#if>
 
     				<#if hasProcedure(data.dispenseResultItemstack)>

```

Comment:
```suggestion
    				<#if hasSuccessCondition>
```

Reply:
hasSuccessCondition refers to the other procedure